### PR TITLE
:book: Update security policy to be specific to OpenSSF Scorecard

### PR DESCRIPTION
**Pull request by justaugustus**
#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?

The [current security policy](https://github.com/ossf/scorecard/blob/5f7cea363724ba03923eb07bdb4e6fa3b2927c15/SECURITY.md) was last updated three years and appears to be boilerplate from Google open source projects.

#### What is the new behavior (if this is a feature change)?**

This updated security policy is specific to the OpenSSF Scorecard project and considers:

- the previously-written policy
- the inherited [SECURITY.md from ossf/.github](https://github.com/ossf/.github/blob/692a363fd99ad0ea50a9512795d53a24e1754d9c/SECURITY.md)
- the LF vulnerability disclosure process for projects: https://www.linuxfoundation.org/security
- the Cisco open source project [SECURITY.md](https://github.com/cisco-ospo/oss-template/blob/main/SECURITY.md), which was itself inspired by the [Wayfair OSS project template](https://github.com/wayfair-incubator/oss-template)

#### Which issue(s) this PR fixes

Partially addresses https://github.com/ossf/scorecard/issues/4194.

#### Special notes for your reviewer

Tagging a few different groups for review here, as the new standard for OpenSSF Scorecard subproject security policies should be something along the lines of:

> This project adheres to the [OpenSSF Scorecard security policy](https://github.com/ossf/scorecard/blob/main/SECURITY.md).

(to minimize drift across the project)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
NONE
```

### CODE REVIEW

**Comment by jeffmendoza** [comment]
Taking a  look at "the inherited [SECURITY.md from ossf/.github](https://github.com/ossf/.github/blob/692a363fd99ad0ea50a9512795d53a24e1754d9c/SECURITY.md)" Should we also mention either:

- security@openssf.org is a fallback
- This policy adheres to the overall policy at https://www.linuxfoundation.org/security

?

File: SECURITY.md [hunk 1]
```diff
@@ -1,11 +1,51 @@
...
+
+The OpenSSF Scorecard maintainers take all security issues in the project
+seriously. Thank you for improving the security of OpenSSF Scorecard. We
+appreciate your dedication to responsible disclosure and will make every effort
+to acknowledge your contributions.
+
+OpenSSF Scorecard leverages GitHub's private vulnerability reporting.
+
+To learn more about this feature and how to submit a vulnerability report,
+review [GitHub's documentation on private reporting](https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing-information-about-vulnerabilities/privately-reporting-a-security-vulnerability).
```
Comment (spencerschrock) [inline SECURITY.md:20]: minor suggestion, removing the `en` allows GitHub to redirect to the reader's preferred locale
```suggestion
To learn more about this feature and how to submit a vulnerability report,
review [GitHub's documentation on private reporting](https://docs.github.com/code-security/security-advisories/guidance-on-reporting-and-writing-information-about-vulnerabilities/privately-reporting-a-security-vulnerability).
```

**Review by spencerschrock — APPROVED** [review]
+1 to Jeff's suggestion on fallback mention. 

I'm otherwise happy with the content, but will leave for others to review

**Comment by spencerschrock** [comment]
> Tagging a few different groups for review here, as the new standard for OpenSSF Scorecard subproject security policies should be something along the lines of:
> 
> > This project adheres to the [OpenSSF Scorecard security policy](https://github.com/ossf/scorecard/blob/main/SECURITY.md).
> 
> (to minimize drift across the project)

Ironically this may cause subprojects to only score a 9/10 for `Security-Policy` based on the last point being awarded for certain terms. (Personally I find that scoring a little too picky but that's how it is currently)

**Comment by justaugustus** [comment]
> Taking a look at "the inherited [SECURITY.md from ossf/.github](https://github.com/ossf/.github/blob/692a363fd99ad0ea50a9512795d53a24e1754d9c/SECURITY.md)" Should we also mention either:
> 
>     * [security@openssf.org](mailto:security@openssf.org) is a fallback
> 
>     * This policy adheres to the overall policy at https://www.linuxfoundation.org/security
> 
> 
> ?

I was thinking the same, but initially decided to leave it out, as I think we should have our own list for this eventually.
That said, it makes to have it here at this stage and I've added that note in https://github.com/ossf/scorecard/pull/4212/commits/1a8adee4ebc49970ec92a148280b3ea87ea25b24.

**Review by jeffmendoza — APPROVED** [review]

**Comment by justaugustus** [comment]
> Ironically this may cause subprojects to only score a 9/10 for `Security-Policy` based on the last point being awarded for certain terms. (Personally I find that scoring a little too picky but that's how it is currently)

@spencerschrock — Good call-out! I've opened https://github.com/ossf/scorecard/issues/4215 so we don't lose that thread.

**Pull request merged**